Composer/packages/client/src/utils/navigation.ts

```diff
@@ -108,7 +110,41 @@ export function convertDialogDiagnosticToUrl(diagnostic: Diagnostic): string {
   return uri;
 }
 
+export function convertInlineLgDiagnosticToUrl(targetTemplateId, dialogs): string {
+  let url = '';
+  function visitor(path: string, value: any): boolean {
+    if (has(value, 'activity') && value.activity === `@{${targetTemplateId}()}`) {
+      url = path;
+      return true;
+    }
+    return false;
+  }
+  dialogs.forEach(dialog => {
+    const triggers = get(dialog, 'content.triggers', []);
+
+    triggers.forEach((t, index) => {
+      searchLgTemplate(`dialogs/${dialog.id}?selected=triggers[${index}]&focused=triggers[${index}]`, t, visitor);
+    });
+  });
+  return url;
+}
+
 export function navigateTo(to: string, navigateOpts: NavigateOptions<NavigationState> = {}) {
   const mapNavPath = resolveToBasePath(BASEPATH, to);
   navigate(mapNavPath, navigateOpts);
 }
+
+function searchLgTemplate(path: string, value: any, visitor) {
```

Comment:
you can use JsonWalk in /lib/indexer.

Reply:
I think we should do it in indexer

Reply:
code changed according to leilei's comments. This comment can be ignored.